src/Playwright.CLI/Program.cs

```diff
@@ -32,21 +32,41 @@ static class Program
     {
         static int Main(string[] args)
         {
-            var root = Directory.GetCurrentDirectory();
+            var path = Directory.GetCurrentDirectory();
             if (args.Length > 1 && args[0] == "-p")
             {
-                root = Path.Combine(Directory.GetCurrentDirectory(), args[1]);
+                path = Path.Combine(Directory.GetCurrentDirectory(), args[1]);
+                var isFile = File.Exists(path);
+                if (!isFile && !Directory.Exists(path))
+                {
+                    return PrintError($"Couldn't find project using Playwright. Ensure a project or a solution exists in {path}, or provide another path using -p.");
+                }
+
+                if (isFile)
+                {
+                    path = Path.Combine(path, "..");
+                }
+
                 args = args[2..];
             }
 
-            var file = Traverse(new DirectoryInfo(root));
+            if (File.GetAttributes(path).HasFlag(FileAttributes.Directory))
+            {
+                var solutions = Directory.GetFiles(path, "*.sln");
+                var projects = Directory.GetFiles(path, "*.*proj");
+                if (solutions.Length == 0 && projects.Length == 0)
+                {
+                    return PrintError($"Couldn't find project using Playwright. Ensure a project or a solution exists in {path}, or provide another path using -p.");
```

Comment:
Nit: it's the same string as above, maybe unify the check/error somewhere?

Reply:
M: Maturity